grammar/liquid-html.ohm

```diff
@@ -91,6 +93,49 @@ LiquidHTML {
   liquidRawTagClose<name> =
     "{%" "-"? space* "end" name space* "-"? "%}"
 
+  // In order for the grammar to "fallback" to the base case, this
+  // rule must pass if and only if we support what we parse. This
+  // implies that—since we don't support filters yet—we have a
+  // positive lookahead on "-}}" or "}}" in the rule. We do this
+  // because we'd otherwise positively match the following string
+  // instead of falling back to the other rule:
+  // {{ 'string' | some_filter }}
+  liquidVariable = liquidExpression space* &("-}}" | "}}") // liquidFilter*
+
+  liquidExpression =
+    | liquidString
+    | liquidNumber
+    | liquidLiteral
+    | liquidRange
+    | liquidVariableLookup
+
+  liquidString = liquidSingleQuotedString | liquidDoubleQuotedString
+  liquidSingleQuotedString = "'" anyExceptStar<("'"| "%}" | "}}")> "'"
+  liquidDoubleQuotedString = "\"" anyExceptStar<("\""| "%}" | "}}")> "\""
+
+  liquidNumber = liquidFloat | liquidInteger
+  liquidInteger = "-"? digit+
+  liquidFloat = "-"? digit+ "." digit+
+
+  liquidLiteral =
+    | "true"
```

Comment:
```suggestion
    | "true" ~variableSegment
```

You should probably guard all of these with a negative lookahead to ensure that they never incorrectly match against a variable name (e.g. `trueRelationship`, `emptyList`, etc.)

See the documentation on [reserved words / keywords](https://ohmjs.org/docs/patterns-and-pitfalls#reserved-words--keywords) in the Patterns and Pitfalls doc.

Reply:
Wow, thanks! I missed that doc and will definitely read it :) Great point about those lookaheads :pray: